### PR TITLE
Attempt fix save script without -includexml (#1609)

### LINKED ISSUE #1609: Save-PSResource script fails
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/Microsoft.PowerShell.PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

Save-PSResource works for modules.

```
PS> Save-PSResource Az
Save-PSResource: The 'file' scheme is not supported.
```

*Note*: That bit about the 'file' scheme not being supported is caused by having a local directory registered as a repository. This is a separate concern and unrelated to the problem here, as I've tested on multiple machines, some without this local repository.

Save-PSResource for scripts fails.

```
PS> Save-PSResource New-OnPremiseHybridWorker
Save-PSResource: The 'file' scheme is not supported.
Save-PSResource: Unable to successfully install package 'New-OnPremiseHybridWorker': 'Could not find file 'C:\Users\***\AppData\Local\Temp\1\2b1ede54-ff18-4731-bfbb-d48de23466df\new-onpremisehybridworker\1.7\New-OnPremiseHybridWorker_InstalledScriptInfo.xml'.'
Save-PSResource: Operation is not valid due to the current state of the object.
Save-PSResource: Package(s) 'New-OnPremiseHybridWorker' could not be installed from registered repositories 'LocalPSGallery, PSGallery'.
```

The multiple repositories is not the cause.  I get the same behavior on other machines with only PSGallery registered. Most of the messages are very poor. In particular "Operation is not valid due to the current state of the object." is developer speak and totally useless to most PowerShell users. The error reporting here should be vastly improved. The first message about the missing XML file does give a slight clue here, however, and the following "works" by some definition.

```
PS> Save-PSResource New-OnPremiseHybridWorker -IncludeXml
Save-PSResource: The 'file' scheme is not supported.
```

This saves the script, but also saves the XML metadata file. I'm not sure why you'd want that file, ever? Regardless, this is different behavior from saving a module. While not fully understanding the desired behavior of these cmdlets (I've posted a question in the discussions, but it appears literally no one reads or at least responds there) I would consider this to be a bug. At best it's inconsistent and unintuitive behavior, making it a usability bug, but I suspect this isn't really the expected behavior.

### Expected behavior

```console
PS> Save-PSResource New-OnPremiseHybridWorker
```

The New-OnPremiseHybridWorker.ps1 script should be saved to the current location without the XML metadata file.
```


### Actual behavior

```console
PS> Save-PSResource New-OnPremiseHybridWorker
Save-PSResource: Unable to successfully install package 'New-OnPremiseHybridWorker': 'Could not find file 'C:\Users\***\AppData\Local\Temp\1\2b1ede54-ff18-4731-bfbb-d48de23466df\new-onpremisehybridworker\1.7\New-OnPremiseHybridWorker_InstalledScriptInfo.xml'.'
Save-PSResource: Operation is not valid due to the current state of the object.
Save-PSResource: Package(s) 'New-OnPremiseHybridWorker' could not be installed from registered repositories 'LocalPSGallery, PSGallery'.
```
```


### Error details

```console
Exception             :
    Type    : Microsoft.PowerShell.PSResourceGet.UtilClasses.ResourceNotFoundException
    Message : Package(s) 'New-OnPremiseHybridWorker' could not be installed from repository 'PSGallery'.
    HResult : -2146233088
TargetObject          : Microsoft.PowerShell.PSResourceGet.Cmdlets.SavePSResource
CategoryInfo          : InvalidData: (Microsoft.PowerShel…lets.SavePSResource:SavePSResource) [Save-PSResource],
ResourceNotFoundException
FullyQualifiedErrorId : InstallPackageFailure,Microsoft.PowerShell.PSResourceGet.Cmdlets.SavePSResource
InvocationInfo        :
    MyCommand        : Save-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 10
    Line             : save-psresource New-OnPremiseHybridWorker -Repository PSGallery
    Statement        : save-psresource New-OnPremiseHybridWorker -Repository PSGallery
    PositionMessage  : At line:1 char:1
                       + save-psresource New-OnPremiseHybridWorker -Repository PSGallery
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : save-psresource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     1.0.3                 Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PS…


Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by o-l-a-v**
# PR Summary

Attempt fix to save script without -includexml, #1609.

Fixes #1609

Test added in PR #1613.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Comment by SydneyhSmith** [comment]
Thanks so much @o-l-a-v  could you please add the tests to this PR so they can all be tested together before merge...thanks!

**Comment by o-l-a-v** [comment]
@SydneyhSmith Done. :)

**Review by anamnavi — APPROVED** [review]
LGTM, thanks for putting this fix in @o-l-a-v  we appreciate it :)

File: test/SavePSResourceTests/SavePSResourceV2Tests.ps1 [hunk 1]
```diff
@@ -173,10 +173,17 @@ Describe 'Test HTTP Save-PSResource for V2 Server Protocol' -tags 'CI' {
         $res.Version | Should -Be "1.0.0.0"
     }
 
+    It "Save script without using -IncludeXML" {
+        Save-PSResource -Name $testScriptName -Repository $PSGalleryName -Path $SaveDir -TrustRepository | Should -Not -Throw
+
+        $SavedScriptFile = [System.IO.Path]::Combine($SaveDir,('{0}.ps1' -f $testScriptName))
```
Comment (anamnavi) [inline test/SavePSResourceTests/SavePSResourceV2Tests.ps1]:
```suggestion
$SavedScriptFile = Join-Path -Path $SaveDir -ChildPath "$testScriptName.ps1"
```

File: test/SavePSResourceTests/SavePSResourceV2Tests.ps1 [hunk 2]
```diff
@@ -173,10 +173,17 @@ Describe 'Test HTTP Save-PSResource for V2 Server Protocol' -tags 'CI' {
...
+        [System.IO.File]::Exists($SavedScriptFile) | Should -BeTrue
```
Comment (anamnavi) [inline test/SavePSResourceTests/SavePSResourceV2Tests.ps1]:
```suggestion
Test-Path -Path $SavedScriptFile | Should -BeTrue
```

**Review by anamnavi — CHANGES_REQUESTED** [review]

**Comment by o-l-a-v** [comment]
I made the requested changes @anamnavi.

**Review by anamnavi — APPROVED** [review]

**Comment by anamnavi** [comment]
Had to approve the workflow to start tests running, once they pass I'll merge it in :)

**Pull request merged**